CorsixTH/Src/persist_lua.cpp

```diff
@@ -406,40 +401,40 @@ class LuaPersistBasicWriter : public LuaPersistWriter
                             lua_checkstack(L, 10);
                             iTable += 2;
                             lua_pushvalue(L, iTable);
-                            lua_pushnumber(L, (lua_Number)(m_iNextIndex++));
+                            lua_pushnumber(L, (lua_Number)(next_index++));
                             lua_settable(L, 2);
                             goto table_reentry; table_resume:
                             iTable -= 2;
                         }
                         else
                         {
                             lua_pop(L, 2);
-                            writeStackObject(-1);
+                            write_stack_object(-1);
                         }
                     }
                     else
-                        writeStackObject(-1);
+                        write_stack_object(-1);
                     lua_pop(L, 1);
                 }
 
                 // Write a nil to mark the end of the children (as nil is the
                 // only value which cannot be used as a key in a table).
                 iType = LUA_TNIL;
-                writeByteStream(&iType, 1);
+                write_byte_stream(&iType, 1);
                 if(iTable != 3)
                     goto table_resume;
                 break; }
 
             case LUA_TFUNCTION:
                 if(lua_iscfunction(L, 2))
                 {
-                    setErrorObject(2);
-                    setError("Cannot persist C functions");
+                    set_error_object(2);
```

Comment:
Is 2 a magic number that should be eliminated?

Reply:
2 is a Lua stack index. There are a lot of magic number lua stack indexes in the code - it makes it very hard to follow.